src/Testing/TestingServiceProvider.php

```diff
@@ -23,7 +23,7 @@ static function (): string {
 
     public function register(): void
     {
-        $this->app->singleton(MockDirective::class);
+        $this->app->bind(MockDirective::class);
```

Comment:
We don't need this binding at all then.

Reply:
Hold on, there is actually a difference in a singleton and static properties. The singleton will be reset when entering new test methods, whereas static properties persist.

Reply:
In this case I can revert back the singleton changes and add 
```
unset($this->directiveArgs);
```
to the BaseDirective::hydrate()

`MockDirective` is the only directive, that can be hydrated more than once.

Reply:
Or I can make `MockResolverService`, that will be a singleton and will be used by `MockDirective` and `mockResolver()`

Reply:
I prefer your second suggestion, it keeps the production code simple and lean.